### ISSUE TITLE
Remove nose dependence

### ISSUE THREAD

**Issue by bdice** [glotzerlab/freud#173]
**[Original report](https://bitbucket.org/glotzer/freud/issue/173) by Bradley Dice (Bitbucket: [bdice](https://bitbucket.org/bdice), GitHub: [bdice](https://github.com/bdice)).**

----------------------------------------

The `nose` package is no longer needed for tests. Some documentation and package requirements still mention `nose`, and should be updated. Personally, I just run `python -m unittest discover .` in the tests directory. This would be one option for changing the docs.

Alternatively, we can continue to require `nose` for CircleCI if we prefer its error outputs over those produced by `unittest`.

**Comment by bdice** [comment]
**Original comment by Bradley Dice (Bitbucket: [bdice](https://bitbucket.org/bdice), GitHub: [bdice](https://github.com/bdice)).**

----------------------------------------

@vyasr Please offer your thoughts sometime.

**Comment by bdice** [comment]
**Original comment by Vyas Ramasubramani (Bitbucket: [vramasub](https://bitbucket.org/vramasub), GitHub: [vyasr](https://github.com/vyasr)).**

----------------------------------------

I agree, I don't use nose either and I don't think we're really leveraging its abilities. To be honest I'm also not convinced that switching to pytest or nose is worthwhile at this stage of the project, at least, it should be a super low priority. We already have largely working tests.

I would also like to remove nose dependence as we continue to clean up the freud dependencies.

**Comment by bdice** [comment]
**Original comment by Bradley Dice (Bitbucket: [bdice](https://bitbucket.org/bdice), GitHub: [bdice](https://github.com/bdice)).**

----------------------------------------

@vyasr I saw you updated this in docs in the the `docs_overhaul` branch. I think that is sufficient. The only other consideration would be if we wanted to move `nose` from `requirements.txt` to a separate `pip install` step in CircleCI. I do like the output of `nosetests` on CI builds, so I'm happy to keep it there.

You may close this issue once you update the README.md file to use `cd tests; python -m unittest discover .`, or we can change the requirements and CI scripts to use nose only for CI builds.

**Comment by bdice** [comment]
**Original comment by Vyas Ramasubramani (Bitbucket: [vramasub](https://bitbucket.org/vramasub), GitHub: [vyasr](https://github.com/vyasr)).**

----------------------------------------

What output do you like? Just the listing on a per-test basis? We can achieve that by just using the verbose argument to `unittest`. We can of course keep using `nose` on CircleCI if you want, but seems sort of unorthodox to me.

**Comment by bdice** [comment]
**Original comment by Bradley Dice (Bitbucket: [bdice](https://bitbucket.org/bdice), GitHub: [bdice](https://github.com/bdice)).**

----------------------------------------

Ooh, good call. I like that. Let's ditch nose.

**Comment by bdice** [comment]
**Original comment by Vyas Ramasubramani (Bitbucket: [vramasub](https://bitbucket.org/vramasub), GitHub: [vyasr](https://github.com/vyasr)).**

----------------------------------------

Cool I'll close this issue once docs is merged then.

**Comment by bdice** [comment]
**Original comment by Bradley Dice (Bitbucket: [bdice](https://bitbucket.org/bdice), GitHub: [bdice](https://github.com/bdice)).**

----------------------------------------

I'll make a PR to remove nose from requirements and alter the CI builds, but I'll leave the doc changes to your PR.

**Comment by bdice** [comment]
**Original comment by Bradley Dice (Bitbucket: [bdice](https://bitbucket.org/bdice), GitHub: [bdice](https://github.com/bdice)).**

----------------------------------------

Docs PR has been merged, and nose is out.
